lib/cn/spill.c

```diff
@@ -239,13 +239,14 @@ cn_subspill(
     ss->ss_added = false;
     ss->ss_work = w;
 
-    if (!sctx->more || key_obj_cmp(&sctx->curr->kobj, &ekobj) > 0) {
-        if (!sctx->pt_set)
-            return 0; /* Nothing to do */
-    }
+    if (!sctx->more)
+        return 0;
 
-    if (!sctx->more || (key_obj_cmp(&sctx->curr->kobj, &ekobj) > 0 && !sctx->pt_set))
-        return 0; /* Nothing to do */
+    /* Proceed only if either the curr key belongs in this leaf node OR there's a ptomb that needs
+     * to be propagated to this child.
+     */
+    if (key_obj_cmp(&sctx->curr->kobj, &ekobj) > 0 && !sctx->pt_set)
```

Comment:
Not that it will make a huge difference, but maybe you want to put the NULL check first since that is going to be a lot faster then a memory comparison.

Reply:
Missed this comment. I'll make this change in my next PR.